Add datepicker

jfrankl opened the pull request:
## Overview

Add datepicker, allowing users to jump to the next or previous timestamp from the detail view.

### Demo

<img width="375" alt="Screen Shot 2022-05-24 at 8 51 43 PM" src="https://user-images.githubusercontent.com/1809908/170155691-5f320322-a7c7-4b51-92f1-405d82398319.png">
<img width="972" alt="Screen Shot 2022-05-24 at 8 51 23 PM" src="https://user-images.githubusercontent.com/1809908/170155695-3888edc2-503d-46ab-ab15-450359454be5.png">
<img width="974" alt="Screen Shot 2022-05-24 at 8 51 30 PM" src="https://user-images.githubusercontent.com/1809908/170155693-da314927-4e4e-4491-9184-24e51dc815c0.png">


### Notes

- I needed to expand the datetime in `getRainfall.js` to be able to split out the date, start time, and end time; as well as use both the full and abbreviated versions of the month


## Testing Instructions

 * How to test this PR
 * Prefer bulleted description
 * Start after checking out this branch
 * Include any setup required, such as bundling scripts, restarting services, etc.
 * Include test case, and expected output


## Checklist

- [ ] `./scripts/lint` runs without errors

Commit:
Add datepicker

KlaasH commented on frontend/data/getRainfall.js:
```diff
@@ -45,10 +45,16 @@ function toLocalTimestamp(isoTimestamp) {
 }
 
-function toDatetimeLabel(isoTimestamp) {
+function toDatetime(isoTimestamp) {
```
`toDatetime` suggests something that will return a `DateTime` object (not capitalizing the T was my mistake earlier), i.e. an instance of the class imported from the luxon package. This is something else (a collection of labels/formats for use in various display contexts) so it's good to include some modifier in the function name to highlight that.  I think `toDateTimeDetails` seems general enough to describe it but specific enough to keep it distinct within the code.

I initially had this as a suggested change, but actually it reaches further down the file and out into the frontend code, so I'm going to make a branch that applies the rename rather than doing it as suggestions throughout the diff.

KlaasH committed:
Rename new more-detailed datetime label to 'dateTimeDetails'

Renames what was the datetime label and is now an object containing a few
different formatted versions of the datetime to make it more clear that
we're talking about that custom object, not an instance of the standard
DateTime class.

Proposed fixup to PR #44, coded up because the diff is big enough that
it would be cumbersome to do it as line-by-line suggestions.

KlaasH reviewed (changes requested):
This looks good!  I had one suggestion, about nomenclature, which turns out to affect enough lines that I did it in code and pushed a modified branch.  It's [commit 9da9ad8c](https://github.com/azavea/sitka-landslide/commit/9da9ad8c382441f8f09702387b4ea3b485d802eb), if you want to just cherry-pick it into your branch. Or I could open a sub-PR against your branch.

jfrankl commented:
@KlaasH This looks great, thank you for your commit. Side note: this was my first time using cherry-pick—so cool!

Made the change. Let me know if you approve and I'll merge this in, or you can feel free to merge it.

KlaasH reviewed (approved):
:+1: Looks good! :cherries:
